src/storage/txn/latch.rs

```diff
@@ -29,34 +30,91 @@ impl Latch {
             waiting: VecDeque::new(),
         }
     }
+
+    /// Find the first command ID in the queue whose hash value is equal to s.
+    pub fn get_first_req_by_hash(&self, s: u64) -> Option<u64> {
+        for item in self.waiting.iter() {
+            if let Some((x, y)) = item {
```

Comment:
if let Some((h, cid)) = item {
  is *h == hash {
    return Some(*cid)
  }
}